resources/vue/src/components/site/common/Pagination.vue

```diff
@@ -0,0 +1,24 @@
+<template>
+    <ul class="pagination mt-5">
+        <li v-for="n in quantity" :key="n" class="page-item">
+            <button class="page-link" @click="changePageListener" :page="n">{{ n }}</button>
+        </li>
+    </ul>
+</template>
+
+<script>
+export default {
+    name: "Pagination",
+    props: ['quantity'],
+    methods: {
+        changePageListener (e) {
+            let page = e.target.attributes.page.value;
+            this.$emit('changePage', page);
+        }
```

Comment:
I think it's ok to mix Composition API with Options for cases like these, but be wary that sometimes teams can enforce only one style of doing things to normalize codebase.